org.osgi.test.common/src/main/java/org/osgi/test/common/context/CloseableBundleContext.java

```diff
@@ -228,13 +228,23 @@ private static boolean delegatedEquals(Object proxy, Object[] args) {
 	}
 
 	public Bundle installBundle(String location, InputStream input) throws BundleException {
-		Bundle bundle = bundleContext.installBundle(location, input);
+		Bundle bundle = bundleContext.getBundle(location);
+		if (bundle != null) {
+			return bundle;
+		} else {
+			bundle = bundleContext.installBundle(location, input);
+		}
 		bundles.add(bundle);
 		return bundle;
 	}
 
 	public Bundle installBundle(String location) throws BundleException {
-		Bundle bundle = bundleContext.installBundle(location);
+		Bundle bundle = bundleContext.getBundle(location);
+		if (bundle != null) {
```

Comment:
Why is this needed? BundleContext.installBundle will return the existing Bundle object if a bundle with the specified location is already installed.

Reply:
not needed. thx removed